packages/server-side-render/src/hook.js

```diff
@@ -0,0 +1,169 @@
+/**
+ * WordPress dependencies
+ */
+import { debounce } from '@wordpress/compose';
+import { useEffect, useState, useRef } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+import { addQueryArgs } from '@wordpress/url';
+import { __experimentalSanitizeBlockAttributes } from '@wordpress/blocks';
+
+export function rendererPath( block, attributes = null, urlQueryArgs = {} ) {
+	return addQueryArgs( `/wp/v2/block-renderer/${ block }`, {
+		context: 'edit',
+		...( null !== attributes ? { attributes } : {} ),
+		...urlQueryArgs,
+	} );
+}
+
+export function removeBlockSupportAttributes( attributes ) {
+	const {
+		backgroundColor,
+		borderColor,
+		fontFamily,
+		fontSize,
+		gradient,
+		textColor,
+		className,
+		...restAttributes
+	} = attributes;
+
+	const { border, color, elements, spacing, typography, ...restStyles } =
+		attributes?.style || {};
+
+	return {
+		...restAttributes,
+		style: restStyles,
+	};
+}
+
+/**
+ * @typedef {Object} ServerSideRenderResponse
+ * @property {string} status  - The current request status: 'idle', 'loading', 'success', or 'error'.
+ * @property {string} [html]  - The rendered HTML content (available when status is 'success').
```

Comment:
Maybe `content` will be a more suitable name here. The `render_callback` can return just string content without any HTML.

Reply:
Updated in [2ea5fa6](https://github.com/WordPress/gutenberg/pull/70543/commits/2ea5fa687b4ef2d5d671ca5b6327ff1f333c61c1).